use constrained quotes when possible

ahus1 opened the pull request:
This tries to fix issue #123. 

The new method FormatAsciiDocAction#isWord() tries to capture Asciidoc's "unconstrained quotes" logic. 

I'm relying on Java 8 `\p{IsAlphabetic}` to identify Unicode characters. 

Comments welcome. I could upload a preview release next week once I am connected to a fast internet again. 

@bodiam, @mojavelinux - please let me know if this looks like what was intended.

mojavelinux commented:
You can consolidate `[\\p{IsAlphabetic}\\p{Digit}_;]` to just `[\\p{L}\\p{Digit}_;]`, which is also more correct. You could even write `(?U)[\\w;]` since Java 7.

You are also missing the colon as a preceding character that requires constrained marks. So the expression for the preceding character to detect the need for constrained formatting should be `[\\p{L}\\p{Digit}_;:]` or  `(?U)[\\w;:]`.

The check for the trailing character should then be: `[\\p{L}\\p{Digit}_]` or `(?U)\\w`.

ahus1 commented:
The colon was not in the user manual - but as I see it works as you describe it in your comment, I sent a pull request. 

Thank you for the concise unicode syntax. I updated the code (and also added some unit test) as part of 93fe1e21100c415011929798d768fe6bc21bc6da.

mojavelinux commented:
:+1:
